spec/rubocop/cop/rspec/repeated_example_group_description_spec.rb

```diff
@@ -0,0 +1,84 @@
+# frozen_string_literal: true
+
+RSpec.describe RuboCop::Cop::RSpec::RepeatedExampleGroupDescription do
+  subject(:cop) { described_class.new }
+
+  it 'registers an offense for repeated describe descriptions' do
+    expect_offense(<<-RUBY)
+      describe 'doing x' do
+      ^^^^^^^^^^^^^^^^^^^^^ Repeated describe block description on line(s) [5]
+        # example group
+      end
+
+      describe 'doing x' do
+      ^^^^^^^^^^^^^^^^^^^^^ Repeated describe block description on line(s) [1]
+        # example group
+      end
+    RUBY
+  end
+
+  it 'registers an offense for repeated context descriptions' do
+    expect_offense(<<-RUBY)
+      context 'when awesome case' do
+      ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ Repeated context block description on line(s) [5]
+        # example group
+      end
+
+      context 'when awesome case' do
+      ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ Repeated context block description on line(s) [1]
+        # example group
+      end
+    RUBY
+  end
+
+  it 'registers an offense with right pointing to lines of code' do
+    expect_offense(<<-RUBY)
+      describe 'super feature' do
+        context 'when some case' do
+          # ...
+        end
+
+        context 'when awesome case' do
+        ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ Repeated context block description on line(s) [10, 14]
+          # example group
+        end
+
+        context 'when awesome case' do
+        ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ Repeated context block description on line(s) [6, 14]
+          # example group
+        end
+
+        context 'when awesome case' do
+        ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ Repeated context block description on line(s) [6, 10]
+          # example group
+        end
+      end
+    RUBY
+  end
+
+  it 'does not register offense for different block descriptions' do
+    expect_no_offenses(<<-RUBY)
+      describe 'doing x' do
+        # example group
+      end
+
+      describe 'doing y' do
+        # example group
+      end
+    RUBY
+  end
+
+  it 'register offense for describe block with additional docstring' do
+    expect_offense(<<-RUBY)
+      RSpec.describe 'Animal', 'dog' do
+      ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ Repeated describe block description on line(s) [5]
+        # example group
+      end
+
+      RSpec.describe 'Animal', 'cat' do
+      ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ Repeated describe block description on line(s) [1]
+        # example group
+      end
+    RUBY
+  end
+end
```

Comment:
And one last case that comes to mind - that it should not flag groups with no descriptions.